edn/src/edn.rustpeg

```diff
@@ -0,0 +1,106 @@
+
+use std::collections::{BTreeSet, BTreeMap, LinkedList};
+use std::iter::FromIterator;
+use types::{Pair, Value};
+use ordered_float::OrderedFloat;
+
+// Goal: Be able to parse https://github.com/edn-format/edn
+// Also extensible to help parse http://docs.datomic.com/query.html
+
+// TODO: Support tagged elements
+// TODO: Support comments
+// TODO: Support discard
+
+#[export]
+nil -> Value = "nil" {
+    Value::Nil
+}
+
+#[export]
+boolean -> Value =
+    "true" { Value::Boolean(true) } /
+    "false" { Value::Boolean(false) }
+
+digit = [0-9]
+sign = "-" / "+"
+
+#[export]
+integer -> Value = i:$( sign? digit+ ) {
+    Value::Integer(i.parse::<i32>().unwrap())
```

Comment:
I'm not good enough at Rust enough to be able to make more than a stab at the right answer here.

The trivial thing would be s/i32/i64/g "because 64 bits should be enough for anyone". But then because it wouldn't, s/i64/[i128](https://github.com/rust-lang/rust/pull/37900)/g or s/i64/[BigInt](https://rust-num.github.io/num/num/struct.BigInt.html)/g which suddenly gets weird and or slow.

We could have the parser auto select the smallest type that fits. Except that it seems awkward if a parse of a well defined data based on a known schema results in different types depending on the data.

My currently thought is that i64 is safe, predictable, easily converted and good enough for 99.9% of cases, and that we should fail to parse for numbers bigger than i64. And that when this becomes a problem we allow global configuration of the number type to use, and when this becomes cumbersome we allow some fancy case-by-case configuration.

Or Rust might do something fancy to help us with this problem that I'm not aware of?